skywalking/agent/__init__.py

```diff
@@ -38,7 +39,8 @@ def __heartbeat():
 def __report():
     while not __finished.is_set():
         if connected():
-            __protocol.report(__queue)  # is blocking actually
+            while __protocol.report(__queue):  # blocking but has timeout
+                pass
```

Comment:
So maybe this is unnecessary after your latest commit? And there is no timeout actually (the comment says it has, which is confusing)

Reply:
You are right on both counts, the `return True` from `Protocol.report()` also becomes unnecessary, stuck around from first commit.